src/components/menu/menu-item/menu-item.component.js

```diff
@@ -75,12 +92,21 @@ const MenuItem = ({
       }
 
       if (submenuContext.handleKeyDown !== undefined) {
-        submenuContext.handleKeyDown(event);
+        /* istanbul ignore else */
```

Comment:
Could you not assert that the `submenuContext.handleKeyDown(event)` isn't called when the item has a search with close icon rather than ignoring it?